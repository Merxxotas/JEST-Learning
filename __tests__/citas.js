import Citas from "../js/classes/Citas";

describe("Probar la clase de Citas", () => {
  const citas = new Citas();

  const id = Date.now();

  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Tobias",
      propietario: "Brayan",
      telefono: "3310027",
      fecha: "10-12-2023",
      hora: "10:59",
      sintomas: "VAGO",
    };

    citas.agregarCita(citaObj);

    // Prueba
    expect(citas).toMatchSnapshot();
  });

  test("actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Nuevo Nombre",
      propietario: "Juan",
      telefono: "19030913",
      fecha: "10-11-2020",
      hora: "10:30",
      sintomas: "Solo duerme",
    };

    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar Cita", () => {
    citas.eliminarCita(id);

    expect(citas).toMatchSnapshot();
  });
});
