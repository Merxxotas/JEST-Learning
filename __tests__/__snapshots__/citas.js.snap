// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "10:59",
      "id": 1692124196862,
      "mascota": "Tobias",
      "propietario": "Brayan",
      "sintomas": "VAGO",
      "telefono": "3310027",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la clase de Citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-11-2020",
      "hora": "10:30",
      "id": 1692124196862,
      "mascota": "Nuevo Nombre",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "19030913",
    },
  ],
}
`;
